File: 0791-custom-sort-string/0791-custom-sort-string.java
class Solution {
    public String customSortString(String order, String str) {
                int[] frequency = new int[26];
        for (int i = 0; i < str.length(); ++i) {
            frequency[str.charAt(i) - 'a']++;
        }
        StringBuilder sortedStringBuilder = new StringBuilder();
        for (int i = 0; i < order.length(); ++i) {
            char currentChar = order.charAt(i);
            while (frequency[currentChar - 'a'] > 0) {
                sortedStringBuilder.append(currentChar);
                frequency[currentChar - 'a']--;
            }
        }
        for (int i = 0; i < 26; ++i) {
            while (frequency[i] > 0) {
                sortedStringBuilder.append((char) ('a' + i));
                frequency[i]--;
            }
        }
        return sortedStringBuilder.toString();
    }
}

        
    

File: 0791-custom-sort-string/0791-custom-sort-string.py
class Solution:
    def customSortString(self, order: str, s: str) -> str:
        char_count = Counter(s)
        sorted_characters = []
        for char in order:
            sorted_characters.append(char * char_count[char])
            char_count[char] = 0
        for char, count in char_count.items():
            sorted_characters.append(char * count)
        return ''.join(sorted_characters)
        